server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -681,6 +681,12 @@ public void updateShardState(
                                 resetEngineToGlobalCheckpoint();
                             }
                             replicationTracker.activatePrimaryMode(getLocalCheckpoint());
+
+                            if (indexSettings.isSegRepEnabled()) {
+                                // force publish a checkpoint now that shard is in primary mode.
+                                checkpointPublisher.publish(this, getLatestReplicationCheckpoint());
```

Comment:
Good catch, the refresh inside of engine reset would not upload anything to remote store bc the shard is not yet in primary mode.  This publish would still hit replicas but it would fetch an older checkpoint from the remote store sync to that until the new primary refreshes.  This is less problematic because the replica would still be able to sync to something where as node-node it is left empty, but it we can avoid this altogether by flushing here to guarantee a refresh occurred, segments were uploaded, and a cp is published.

Reply:
@ankitkala Took a look here to add a flush/refresh and force upload and replication.  This actually isn't required because the primary would only upload a new commit pt with no actual segments.  With remote store this publish would force a replica to check if it is up to date, given pterm is higher it would trigger a round of segrep.  It would go to the store and fetch the latest checkpoint which if nothing is uploaded yet from new primary would be from the previous pterm.  Once the new primary uploads something to the store it will still correctly sync.  This flow is covered with testDropPrimaryDuringReplication.